recipes/libgcrypt/all/conanfile.py

```diff
@@ -0,0 +1,84 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.33.0"
+
+class LibgcryptConan(ConanFile):
+    name = "libgcrypt"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.gnupg.org/download/index.html#libgcrypt"
+    description = "Libgcrypt is a general purpose cryptographic library originally based on code from GnuPG"
+    topics = ("conan", "libgcrypt", "gcrypt", "gnupg", "gpg", "crypto", "cryptography")
+    license = "LGPL-2.1-or-later"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+    generators = "cmake"
+    requires = 'libgpg-error/1.36'
+    exports_sources = "CMakeLists.txt", "config.h.in", "gcrypt.h.in", "mod-source-info.h", "patches/**"
+
+    _cmake = None
+
+    @property
+    def _is_msvc(self):
+        return self.settings.compiler == "Visual Studio"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("This recipe only support Linux. You can contribute Windows and/or Macos support.")
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if self.options.shared:
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        if self._is_msvc:
+            for patch in self.conan_data["patches"][self.version]:
+                tools.patch(**patch)
+            os.makedirs(os.path.join(self._source_subfolder, "sys"))
+            tools.download("https://raw.githubusercontent.com/win32ports/unistd_h/master/unistd.h",
+                           os.path.join(self._source_subfolder, "unistd.h"))
+            tools.download("https://raw.githubusercontent.com/win32ports/sys_time_h/master/sys/time.h",
+                           os.path.join(self._source_subfolder, "sys", "time.h"))
```

Comment:
Sources should be immutable

Reply:
needed only for win32, can be removed for linux-only build

Reply:
How about creating proper `win32ports-unistd-h`/`win32-ports-sys_sime` conan recipes and using those?

Reply:
yes, it will be needed for win32 build, eventually (not right now)

Reply:
Indeed, let's make it a `FIXME: create + use win32ports-unistd-h/win32-ports-sys_sime for Windows/MSVC support`.